include/linux/ancillary_bus.h

```diff
@@ -0,0 +1,61 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+/*
+ * Copyright (c) 2019-2020 Intel Corporation
+ *
+ * Please see Documentation/driver-api/ancillary_bus.rst for more information.
+ */
+
+#ifndef _ANCILLARY_BUS_H_
+#define _ANCILLARY_BUS_H_
+
+#include <linux/device.h>
+#include <linux/slab.h>
+
+struct ancillary_device {
+	struct device dev;
+	const char *name;
+	const char *match_name;
+	u32 id;
+};
+
+struct ancillary_driver {
+	int (*probe)(struct ancillary_device *adev);
+	int (*remove)(struct ancillary_device *adev);
+	void (*shutdown)(struct ancillary_device *adev);
+	int (*suspend)(struct ancillary_device *adev, pm_message_t state);
+	int (*resume)(struct ancillary_device *adev);
+	struct device_driver driver;
+	const struct ancillary_device_id *id_table;
+};
+
+static inline struct ancillary_device *to_ancillary_dev(struct device *dev)
+{
+	return container_of(dev, struct ancillary_device, dev);
+}
+
+static inline struct ancillary_driver *to_ancillary_drv(struct device_driver *drv)
```

Comment:
@dmertman 
I missed this. these macros are used only in the ancillary_bus.c, so lets move it there.

Reply:
is there a possibility that the parent drivers might need these?

Reply:
@paravmellanox we use the to_ancillary_dev() macro in the sof-client.c file. For the sake of consistency can we keep all the to_ancillary_*() macros in the header?

Reply:
@ranj063 if it is used, it is already at good place. no need to move.
thanks.